Source/WebCore/dom/Document.cpp

```diff
@@ -2304,13 +2305,13 @@ bool Document::updateStyleIfNeeded()
     return true;
 }
 
-void Document::updateLayoutIgnorePendingStylesheets(OptionSet<LayoutOptions> layoutOptions)
+void Document::updateLayoutIgnorePendingStylesheets(OptionSet<LayoutOptions> layoutOptions, const Element* context)
 {
     layoutOptions.add(LayoutOptions::IgnorePendingStylesheets);
-    updateLayout(layoutOptions);
+    updateLayout(layoutOptions, layoutOptions.contains(LayoutOptions::ContentVisibilityForceLayout) ? context : nullptr);
```

Comment:
Passing the context unconditionally here probably doesn't hurt. It could be used for different optimizations unrelated to content visibility

Reply:
You are right, cleaner that way, adjusted.